tinygrad/engine/schedule.py

```diff
@@ -206,6 +205,7 @@ def create_schedule_with_vars(outs:List[LazyBuffer], seen:Optional[Set[LazyBuffe
   graph: DefaultDict[LazyBuffer, List[LazyBuffer]] = defaultdict(list)
   in_degree: DefaultDict[LazyBuffer, int] = defaultdict(int)
   for key, lsi in prescheduled.items():
+    if key not in in_degree: in_degree[key] = 0
```

Comment:
Hmm, so defaultdict should do this, but I see why there's reasons you want to do it explicitly (for iteration over keys later). Do you still need the defaultdict with this line, or can you make in_degree a normal dict?

Reply:
`assign_targets` needs it, you could change `in_degree[assign]` without having visited the node
```py
    # realize outputs before a parent is assigned to
    parents_assigns = set(schedule_targets[assign_targets[x]].outputs[0] for x in lsi.inputs if x in assign_targets)
    for assign in parents_assigns:
      graph[key].append(assign)
      in_degree[assign] += 1
```